### ISSUE TITLE
Allow unparenthesized expressions to contain more than one comparison operator

### ISSUE THREAD

**Issue by andrew-skybound** [sass/libsass#1727]
Ruby SASS allows multiple comparison operators to appear in a single unparenthesized expression, in the format:

```
	a == b == c
	a > b == c
```

And so on.  Indeed "c" needs to be a boolean value for this to be useful but it is nevertheless supported.  Libsass currently only allows one comparison operator (`== != > >= < <=`) per unparenthesized expression.

Input:
```SCSS
$x: 2;
$y: 1;

@function getResult() {
	@return true;
}

.test {
	a: $x > $y == getResult();
}
```

Ruby SASS Output (3.4.19):
```CSS
.test {
  a: true;
}
```

Libsass Output (3.2.5):
```CSS
Error: error reading values after $y
        on line 9 of test.scss
>>      a: $x > $y == getResult();
   -----------^
```

**Comment by xzyfer** [comment]
Spec added https://github.com/sass/sass-spec/pull/603

**Comment by mgreter** [comment]
This was one of the major fixes in the milestone 3.4 branch and I've added a pretty extensive test suite to back up this feature! Code for it is a bit dirty in some places, but specs do pass!
